src/sparseml/transformers/integration_helper_functions.py

```diff
@@ -125,39 +124,42 @@ def create_model(
             config=config,
             split="validation",
         )
-        if task in TaskNames.text_generation.value and not dataset_with_labels:
-            validation_dataset = validation_dataset.remove_columns("labels")
 
-    trainer = initialize_trainer(model, source_path, validation_dataset)
+        if task in TaskNames.text_generation.value:
+            # text-generation datasets have a separate
+            # logic for creating a dataloader
+            if not dataset_with_labels:
+                validation_dataset = validation_dataset.remove_columns("labels")
+            data_loader = format_calibration_data(tokenized_dataset=validation_dataset)
```

Comment:
I'm all good with using this function more generally, but I think we should probably change the name to something like 'format_dataloader_from_dataset' to make it less specific to calibration data